js-exercises/permutate-without-repetitions/permutateWithoutRepetitions.js

```diff
@@ -2,6 +2,19 @@
  * @param {*[]} permutationOptions
  * @return {*[]}
  */
-function permutateWithoutRepetitions(permutationOptions) {}
+function permutateWithoutRepetitions(array, permutationSet = [], result = []) {
```

Comment:
What is the relevance of permutationSet = [], result = [] parameters ?

Reply:
Here `permutationWithoutRepetitions` is a recursive function. Which `result `parameter stores all the computed permutation in an array.  And `permutationSet` contains all visited element of provided array. and visited element is removed from `array`, Now in which recurive call array gets empty .We push permutationSet to result array.

Basically DFS is implemented using recursion.